notebooks/amazon_forest_notebook.py

```diff
@@ -190,119 +191,121 @@
+
+model = Model(input_tensor, output)
 
-batch_size = 64
-epochs_arr = [35, 15, 5]
-learn_rates = [0.002, 0.0002, 0.00002]
+model.summary()
 
 # <markdowncell>
 
-# ## Define and Train model
-# 
-# Here we define the model and begin training. 
+# ## Fine-tune conv layers
+# Now that we have a trained classifier, we can unfreeze all other layers in the model (or keep certain layers still frozen) and retrain. 
 
 # <codecell>
 
-classifier = AmazonKerasClassifier(preprocessor)
-classifier.add_conv_layer()
-classifier.add_flatten_layer()
-classifier.add_ann_layer(len(preprocessor.y_map))
+from keras.optimizers import Adam
+from keras.callbacks import ModelCheckpoint, EarlyStopping, ReduceLROnPlateau
 
-train_losses, val_losses = [], []
-for learn_rate, epochs in zip(learn_rates, epochs_arr):
-    tmp_train_losses, tmp_val_losses, fbeta_score = classifier.train_model(learn_rate, epochs, batch_size, 
-                                                                           train_callbacks=[checkpoint, fbeta, csv])
-    train_losses += tmp_train_losses
-    val_losses += tmp_val_losses
+callbacks = [EarlyStopping(monitor='val_loss', patience=2, verbose=1, min_delta=1e-4),
+             ReduceLROnPlateau(monitor='val_loss', factor=0.5, patience=1, cooldown=0, verbose=1),
+             ModelCheckpoint(filepath='weights/weights.best.hdf5', verbose=1, save_best_only=True, 
+                             save_weights_only=True, mode='auto')]
 
-print(fbeta.fbeta)
+X_train, y_train = preprocessor.X_train, preprocessor.y_train
+X_val, y_val = preprocessor.X_val, preprocessor.y_val
 
-# <markdowncell>
+epochs_arr = [5, 3, 2]
+learn_rates = [1e-4, 1e-5, 1e-6]
+batch_size = 128
+train_generator = preprocessor.get_train_generator(batch_size)
+steps = len(X_train) // batch_size
 
-# ## Load Best Weights
+for learn_rate, epochs in zip(learn_rates, epochs_arr):
+    model.compile(optimizer=Adam(lr=learn_rate), loss='binary_crossentropy', metrics = ['accuracy'])
+    model.fit_generator(train_generator, steps, epochs, verbose=1, 
+                        validation_data=(X_val, y_val), callbacks=callbacks)
 
 # <markdowncell>
 
-# Here you should load back in the best weights that were automatically saved by ModelCheckpoint during training
+# ## Load Best Weights
 
 # <codecell>
 
-classifier.load_weights("weights.best.hdf5")
+model.load_weights("weights/weights.best.hdf5")
 print("Weights loaded")
 
 # <markdowncell>
 
-# ## Monitor the results
+# ## Check Fbeta Score
 
-# <markdowncell>
+# <codecell>
 
-# Check that we do not overfit by plotting the losses of the train and validation sets
+from sklearn.metrics import fbeta_score
 
-# <codecell>
+def get_fbeta_score(model, X_valid, y_valid):
+    p_valid = model.predict(X_valid)
+    return fbeta_score(y_valid, np.array(p_valid) > 0.2, beta=2, average='samples')
 
-plt.plot(train_losses, label='Training loss')
-plt.plot(val_losses, label='Validation loss')
-plt.legend();
```

Comment:
I honestly didn't find much use for it and because I was dealing with limited memory allocation I didn't want to waste any memory where it wasn't absolutely essential for training (i.e. so we can use a bigger batch size). If you disagree then I can add the history callback back in including the plot.